### ISSUE TITLE
Provide a way to mark experiments as unsuitable for DEA

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#162]
This would be a curator-assigned flag, probably implemented as an auditevent like "UnsuitableForDiffExDeterminationEvent".

Example: https://gemma.msl.ubc.ca/expressionExperiment/showExpressionExperiment.html?id=18038  (GSE159778). This looked like it might be usable if we subset, but then the only two experimental factors are confounded in each of the "Experiments". Splitting this experiment won't help either because of the confound.

It's possible this is a curation issue in this specific case, but there are going to be experiments where curators will keep looking at them to figure out how to do the DEA, when the answer is that it can't. 

Curators should be able to see if that's been determined, and the same flag can be used by the system to not do DEA on an experiment  (unless -force) - this will make bulk reprocessing skip them for example.

**Comment by ppavlidis** [comment]
There should probably be a way to reverse this event, which makes the audit trail less suitable. Though deleting audit events is possible, we don't like to do it as it tracks the history; unless we break that rule, adding a second event like "SuitableForDiffExDeterminationEvent" would be needed. The alternative which is simple is a field in the EE itself, but it loses functionality of the audit event like attaching a note. I would prefer the audit trail way overall.

**Comment by AlexAdrian-Hamazaki** [comment]
I totally agree. Currently we are just putting the experiments as public and writing admin notes. However, the public isn't able to see the notes, so it may be confusing. 

Let us now if you want some more examples or if we can help in any way

**Comment by ppavlidis** [comment]
Thanks. It's straightforward, no examples needed, but there are several pieces of implementation to do:

* UnsuitableForDiffExDeterminationEvent - "Indicates that it was determined the experiment is not suitable for differential expression due to the experimental design"
* Add UI mechanism to add this event, probably on the admin tab next to the DEA button or the usability status button - must have a dialog box to allow entering the explanation, controller method on back end
* UI mechanism for reversion (lower priority)
* Code to add badge to experiment web pages, and this status might have to be part of the ExpressionExperimentValueObject, which also allows us to expose to the API

**Comment by ppavlidis** [comment]
- [ ] UnsuitableForDiffExDeterminationEvent - "Indicates that it was determined the experiment is not suitable for differential expression due to the experimental design"
- [ ] Add UI mechanism to add this event, probably on the admin tab next to the DEA button or the usability status button - must have a dialog box to allow entering the explanation, controller method on back end
- [ ] UI mechanism for reversion (lower priority)
- [ ] Code to add badge to experiment web pages, and this status might have to be part of the ExpressionExperimentValueObject, which also allows us to expose to the API

**Comment by ppavlidis** [comment]
For now the status is shown on the admin tab, and this status is manipulated the same way we do for "usability" - it's a pair of new options.

I have confirmed that this works, but we have to be careful testing this on dev.gemma because it introduces a data model change in the form of new classes that Hibernate has to identify (the new Event types). Inserting new types of auditevents into the db behind hibernate's back can cause the production interface to fail as when they are retrieved Hibernate doesn't know what they are. (I may have previously implemented a way to avoid those errors, but I can't find it now so can't confirm that is in place).

In other words, we'll have to roll this to production to test it properly, but further development can carry on after the production instance is "aware" of the new event types.

**Comment by ppavlidis** [comment]
I am not sure if we want to add a badge for this for regular users to see, but we'll do that after we confirm the basic curation mechanism works.
